contracts/src/Teleporter/upgrades/README.md

```diff
@@ -0,0 +1,62 @@
+# Upgradeability
+
+## Overview
+
+The `TeleporterMessenger` contract is non-upgradable, once a version of the contract is deployed it cannot be changed. This is with the intention of preventing any changes to the contract that could potentially introduce bugs or vulnerabilities.
+
+However, we do want to be able to provide new features and improvements to the `TeleporterMessenger` contract. To do this we have implemented a `TeleporterRegistry`.
+
+The `TeleporterRegistry` keeps track of a mapping of `TeleporterMessenger` contract versions to their addresses. This allows us to deploy new versions of the `TeleporterMessenger` contract and then update the `TeleporterRegistry` to point to different versions. The `TeleporterRegistry` can only be updated through a Warp out-of-band message that qualifies the following requirements:
```

Comment:
discussed offline, we are going to stick with "version", since "nonce" is also an overloaded term. Developers might think nonces should only be incremental, but for the registries old versions can be added. Added clarification that version mapping is independent across blockchains, and up to chain to choose their mapping.